dev-docs/RFCs/v8.0/luma-refactor.md

```diff
@@ -0,0 +1,83 @@
+- Reduce its complexity and associated maintenance costs
+- Reduce the weight it imposes on applications that use it
+- Make its value proposition clear to potential users
+- Simplify adoption
+
+## Overview
+
+The first step towards simplifying luma.gl is to be explicit about what its goals are. I propose that changes be made around the following key principles:
+- Targeting intermediate to advanced WebGL users
+- All systems implemented should solve explicitly identified problems. Experiments are encouraged, but those that fail to find a use case should be removed
+- Structured around levels of abstraction, primarily helper tools for low-level WebGL work, WebGL wrapper classes, and higher-level engine features
+- Increased focus on architectural fundamentals, simplicity, building systems that are easy to reason about. Less focus on using the latest APIs or frameworks
+
+
+The actual refactor of luma.gl would occur in three phases:
+
+### Clean Out Unused Code
+
+This will warrant some discussion. The following are systems I believe should be considered for removal or consolidation:
+- [ModelNode](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/scenegraph/nodes/model-node.js), [BaseModel](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/base-model.js), [Model](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/model.js) - functionality could be supported with just the `Model` and `Group` classes
+- [CameraNode](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/scenegraph/nodes/camera-node.js) - Appears unused
+- [Uniform animations](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/base-model.js#L280-L333) - Complex and unused
+- [ShaderCache](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/shader-cache.js) - Superseded by `ProgramManager`
+- [core/lighting](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lighting/light-source.js), [core/materials](https://github.com/uber/luma.gl/tree/7.2-release/modules/core/src/materials) - currently empty or simple data classes
+- [core/multipass](https://github.com/uber/luma.gl/tree/7.2-release/modules/core/src/multipass) - Appears unused (has it been superseded by deck effects?)
+- [seer integration](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/debug/seer-integration.js) - Complex and will interfere with other parts of the system if enabled (e.g. [GPU timing](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/base-model.js#L337-L376)). Is anyone using it?
+- [debug/parameter-definitions](https://github.com/uber/luma.gl/blob/7.2-release/modules/debug/src/webgl-api-tracing/parameter-definitions.js) - Appears unused
+- [main module](https://github.com/uber/luma.gl/tree/7.2-release/modules/main) - Appears unused
```

Comment:
Yes I think we should cut the main modules from all libs except deck, for history reasons.